src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -412,10 +412,10 @@ private async Task<bool> FindFilesAsync(TOptions options, TContext rootContext)
                     }
 #endif
 
-                    foreach (string file in FileSystem.DirectoryEnumerateFiles(directory,
-                                                                               filter,
-                                                                               SearchOption.TopDirectoryOnly))
+                    foreach (string file in FileSystem.DirectoryEnumerateFiles(directory, filter, SearchOption.TopDirectoryOnly)
+                            .Where(file => IsTargetWithinFileSizeLimit(file, _rootContext.FileSizeInKilobytes)))
```

Comment:
>Where [](http://example.com/codeflow?start=29&length=5)

Please put your predicate/handling within the collection, the rule is to avoid chained collection traversal.